app/views/shared/_header.html.erb

```diff
@@ -74,8 +74,10 @@
             <li><%= link_to "Secrets", admin_secrets_path %></li>
           <% end %>
           <li><%= link_to "Reports", csv_exports_path %></li>
-          <li><%= link_to "OAuth Applications", oauth_applications_path %></li>
-          <li><%= link_to "OAuth Authorized Applications", oauth_authorized_applications_path %></li>
+          <% if current_user.try(:super_admin?) %>
```

Comment:
ruby 2.3 means you can use the lonely operator now :D `current_user&.super_admin?`